x-pack/plugins/enterprise_search/public/applications/app_search/components/engine_overview/engine_overview.tsx

```diff
@@ -32,9 +32,7 @@ export const EngineOverview: React.FC = () => {
     pollForOverviewMetrics();
   }, []);
 
-  if (dataLoading) {
-    return <Loading />;
-  }
+  if (dataLoading) return <AppSearchPageTemplate isLoading />;
```

Comment:
I'm not a huge fan of this particular whole-page `isLoading` early return, but unfortunately with pages that have page titles jump change based on loaded data (in this case, we show either `Engine overview` or `Engine setup` depending on whether the engine has documents), it's better not to show the page at all initially - otherwise what happens is you get a brief flash of the `Engine setup` page which then jumps to `Engine overview` 🤷 

WS has similar issues with their overview as well. Discussion for more context: https://github.com/elastic/kibana/pull/102592#discussion_r654544883

In this case, my ideal solution for this view would be to abstract the documentsCount / isEngineEmpty check to the top level `engine` logic available to all engine pages. That would allow the setup vs overview check to happen instantaneously, no data loading needed. I didn't want to add more lines into this PR though, so I left it out for now - time permitting, I'll follow up with some UX polish PRs this week.

Reply:
Yeah I'd like to have a larger conversation about loading and how we handle it, I think we could improve the consistency of our loading patterns across the app for section routers, individual views, and individual componnennts. I agree it was better to limit this PRs scope

Reply:
The new page template does actually improve consistent page-level loading patterns fairly significantly! I like that the page headers load in instantaneously, it makes navigation feel much snappier - it just backfires in scenarios where we conditionally change the entire page header/title heh (which we should consider potentially moving away from as a UX pattern).

Like you said though, there's always room to improve and think about loading individual components instead 👍